ddtrace/contrib/pytest/plugin.py

```diff
@@ -191,7 +191,11 @@ def pytest_runtest_makereport(item, call):
                 # XPass (strict=True) are recorded failed by pytest, longrepr contains reason
                 span.set_tag(test.XFAIL_REASON, result.longrepr)
                 span.set_tag(test.RESULT, test.Status.XPASS.value)
-            raise RuntimeWarning(result)
```

Comment:
I'm probably missing something, but I see no reason to raise an exception here. We can use `call.excinfo` directly to set the exception info to the span.

Reply:
I wonder it is is force the `except Exception:` code?

Reply:
yeah I guess

Reply:
yup looks like it was raised to be caught below to set the span tags - so we can get rid of that logic now

Reply:
(unless `outcome.get_result()` is expected to raise on test failure which would seem wrong)

Reply:
yeah I think it shouldn't raise, so you're right, we can get rid of that logic